kernel/src/snapshot.rs

```diff
@@ -385,6 +400,48 @@ impl Snapshot {
 
         domain_metadata_configuration(self.log_segment(), domain, engine)
     }
+
+    /// Get the In-Commit Timestamp for this snapshot, if available.
+    /// It returns the inCommitTimestamp from the CommitInfo action of the commit that created this snapshot.
+    ///
+    /// Returns:
+    /// - `Ok(Some(timestamp))` if ICT is enabled and available for this snapshot
+    /// - `Ok(None)` if ICT is not enabled for this table, or if this snapshot predates ICT enablement
+    /// - `Err(...)` if ICT is enabled but there was an error reading it
+    pub fn get_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<Option<i64>> {
+        // Check if we have a cached value first
+        if let Some(cached_timestamp) = self.latest_in_commit_timestamp.get() {
+            return Ok(*cached_timestamp);
+        }
+
+        let enablement_info = self
+            .table_configuration()
+            .in_commit_timestamp_enablement()?;
+
+        match enablement_info {
+            Some((enablement_version, _)) if self.version() >= enablement_version => {
+                // For versions >= enablement_version, ICT must be present
+                let commit_path = ParsedLogPath::new_commit(self.table_root(), self.version())?;
```

Comment:
I think this will break for catalog-managed. Do we have a better way of deducing the path we need to read for ICT? perhaps from logsegment?

Reply:
Curious to know if the catalog will always be returning this information?

Reply:
For filysestem, kernel must either look at the previous commit or the previous CRC for this info. So I think it would be safe to block this for catalog-managed, and flesh out the ICT story for CCV2 in the future.